spk/owncloud/src/service-setup.sh

```diff
@@ -0,0 +1,487 @@
+                    files=$(find "$source" -type f -name "$pattern")
+                    if [ -n "$files" ]; then
+                        for file in "${files[@]}"; do
+                            ${CP} "$file" "$target/"
+                        done
+                    fi
+                done
+            fi
+
+            # Check user data export
+            if [ "${wizard_export_userdata}" = "true" ]; then
+                echo "Copying previous user data from ${DATADIR}"
+                ${CP} "${DATADIR}" "${TEMPDIR}/"
+            fi
+
+            # Create backup archive
+            archive_name="$(basename "$TEMPDIR").tar.gz"
+            echo "Creating compressed archive of ownCloud data in file $archive_name"
+            /bin/tar -czvf "${SYNOPKG_PKGTMP}/$archive_name" "$TEMPDIR" 2>&1
+
+            # Move archive to export directory
+            if ${RSYNC} "${SYNOPKG_PKGTMP}/$archive_name" "${wizard_export_path}/"; then
+                echo "Backup file copied successfully to ${wizard_export_path}."
+            else
+                echo "File copy failed. Trying to copy to alternate location..."
+                if ${RSYNC} "${SYNOPKG_PKGTMP}/$archive_name" "${SYNOPKG_PKGDEST_VOL}/@tmp/"; then
+                    echo "Backup file copied successfully to alternate location (${SYNOPKG_PKGDEST_VOL}/@tmp)."
+                else
+                    echo "File copy failed. Backup of ownCloud data will not be saved."
+                fi
+            fi
+
+            # Clean-up temporary files
+            ${RM} "${TEMPDIR}"
+            ${RM} "${SYNOPKG_PKGTMP}/$archive_name"
+        fi
+    fi
+}
+
+service_postuninst ()
+{
+    if [ ${SYNOPKG_DSM_VERSION_MAJOR} -lt 7 ]; then
+        # Remove the web interface
+        ${RM} ${OCROOT}
+
+        # Remove web configurations
+        TEMPDIR="${SYNOPKG_PKGTMP}/web"
+        ${MKDIR} ${TEMPDIR}
+        WS_CFG_PATH="/usr/syno/etc/packages/WebStation"
+        PHP_CFG_FILE="PHPSettings.json"
+        WS_TMPL_PATH="/var/packages/WebStation/target/misc"
+        WS_TMPL_FILE="php74_fpm.mustache"
+        CFG_UPDATE="no"
+        # Check for ownCloud PHP profile
+        if ${JQ} -e '.["com-synocommunity-packages-owncloud"]' "${WS_CFG_PATH}/${PHP_CFG_FILE}" >/dev/null; then
+            echo "Removing PHP profile for ownCloud"
+            ${JQ} 'del(.["com-synocommunity-packages-owncloud"])' ${WS_CFG_PATH}/${PHP_CFG_FILE} > ${TEMPDIR}/${PHP_CFG_FILE}
+            ${MV} ${WS_CFG_PATH}/${PHP_CFG_FILE} ${WS_CFG_PATH}/${PHP_CFG_FILE}.bak
+            rsync -aX ${TEMPDIR}/${PHP_CFG_FILE} ${WS_CFG_PATH}/ 2>&1
+            ${RM} ${TEMPDIR}/${PHP_CFG_FILE}
+            CFG_UPDATE="yes"
+        fi
+        # Check for backup PHP template
+        if [ -f "$WS_TMPL_PATH/$WS_TMPL_FILE.bak" ]; then
+            echo "Restore PHP template from backup"
+            ${RM} ${WS_TMPL_PATH}/${WS_TMPL_FILE}
+            ${MV} ${WS_TMPL_PATH}/${WS_TMPL_FILE}.bak ${WS_TMPL_PATH}/${WS_TMPL_FILE}
```

Comment:
Two questions (may be naive):

1. Since `${MV}` is a shortcut for `mv -f` I suspect the initial rm is redundant, right ?
2. Wouldn't this principle of "restoring" former configuration file be fragile in case of other packages using the same mechanisms ? (thinking about tt-rss here because I would be glad to use the same tricks)

Reply:
Thanks for your feedback. You are correct that the `${RM}` may be redundant but I guess I was just trying to be safe. As for the config restoration, you are correct in that it may be fragile. I'm going to try to make it a bit more robust.

Reply:
At worse you could leave that to the next (there is already a lot of work in this MR).

Personally I would only have shown a message "post install" to indicate that some settings were changed and leave those untouched when uninstalling (apart from the PHP profile removal, that is).

Reply:
I've changed the restoration of the configuration explicitly change the new string back to the old string using `sed` so it doesn't rely on the presence of a backup file.